msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java

```diff
@@ -1499,7 +1599,21 @@ public void markMessageAsRepliedForUser(final PrivateMessage message) {
 				}
 			}
 		}		
-	} 
+	}
+
+	public void markMessageAsRepliedForUser(final PrivateMessage message, String userId) {				
```

Comment:
This method could be reduced to 2 lines:
```
Optional<List<PrivateMessageRecipient>> recipientList = Optional.ofNullable(getPrivateMessageWithRecipients(message).getRecipients());
recipientList.map(Collection::stream).orElseGet(Stream::empty).filter(r -> r.getUserId().equals(userId)).forEach(r -> r.setReplied(true));
```

Reply:
This method is an overloaded version of an existing one in which the functionality is very similar, modifying the coding may lead to confusion of the implementation of the same.